### PR TITLE
React Native Android SDK Checks

### LINKED ISSUE #17: Support for React Native Android
Originally this ticket:  https://github.com/infinitered/solidarity-react-native/issues/4
Then added code to envinfo:
Once this is merged: https://github.com/tabrindle/envinfo/pull/16

Add support for detecting and handling specific Android versions in this plugin.

### PR DESCRIPTION
**Pull request by GantMan**
## TL:DR;
This plugin now makes sure your project and your Android SDKs are good.

### Solidarity Check:  With Android Rule
checks the React Native API and BuildTools to make sure they exist on the system.

### Snapshot:  Not needed
Since the version is identified in the RN project, no need to change or modify the rule versions.  This makes life simple for everyone.

Implements custom rule for Android SDK version checks on API/Build Tools

### Report:
Identifies RN project API and BuildTool version, and if it exists on the machine
screen shot 2018-02-14 at 9 44 45 pm

Availability check thanks to incoming API from envinfo

### Rule example:
```json
    "Android SDK Check": [
      {
        "rule": "custom",
        "plugin": "React Native",
        "name": "androidVersion"
      }
    ]
```

-------
closes #17

### CODE REVIEW

**Comment by GantMan** [comment]
Note this requires that:
A stable version of EnvInfo is released, followed by a Solidarity 2.0.0 (peer dependency of this plugin).
